### ISSUE TITLE
List index out of error

### ISSUE THREAD

**Issue by pachax001** [Prudhvi-pln/udb#5]
Try this drama https://myasiantv.ac/show/temptation-of-wife/
When try to download episode 7 get an error saying list out of index. Because that episode is coming through streamsb.
Can you apply a method to skip these types of links or download them? Also ifthe episode download is skipped show the skipped episode in a txt file or in a log

**Comment by Prudhvi-pln** [comment]
This involves 3 fixes/enhancements:
1. List out of index - BUG - Fixed
2. Show skipped episodes in log - ENHANCEMENT - In-Progress

Issue-1 was addressed as part of my latest commit. But there is another issue where the default link is very slow and often causing timeout. This requires to try with an alternate source, which will be planned as an enhancement.
For now, you can blacklist the slow links in the configuration.

**Comment by Prudhvi-pln** [comment]
Added skipped episode info as part of summary in v2.9.0
```
Download Summary:
[2023-10-05 21:49:02] Download completed for Radio Star Episode 436 - 360P.mp4 in 01m 13s!
[2023-10-05 21:47:49] Download skipped for Radio Star Episode 436.1 - 100P.mp4 due to error: ID not found in stream link
[2023-10-05 21:51:16] Download completed for Radio Star Episode 437 - 360P.mp4 in 03m 27s!
```

**Comment by pachax001** [comment]
This links also doesnt show up https://asianhd2.cc/download?id=NDA3NDQ=&typesub=SUB&title=The+Legendary+Witch+Episode+18
Try legend of the witch in drama section and when try to download episode link gathering stops from episode 11

**Comment by Prudhvi-pln** [comment]
I found the issue behind this. The problem is that some episodes are exposed directly as mp4 files instead of m3u8. I added a check to skip non-m3u8 links for now. Please refer v2.9.1: https://github.com/Prudhvi-pln/udb/commit/1f641506b0ca2743dfa8d0c986ecaa66033ba2a7 

I'm working on ability to download non-m3u8 links.

**Comment by Prudhvi-pln** [comment]
@pachax001, this issue is addressed as part of release [v2.10.0](https://github.com/Prudhvi-pln/udb/releases/tag/v2.10.0). Please check and re-open this if the issue still exists.
